src/Service/FeedService.php

```diff
@@ -6,41 +6,46 @@
 use ApiPlatform\Core\Api\UrlGeneratorInterface;
 use ApiPlatform\Core\Bridge\Symfony\Routing\RouteNameGenerator;
 use App\Entity\Feed;
-use App\Event\GetFeedTypesEvent;
-use App\Feed\FeedTypeInterface;
 use Symfony\Component\EventDispatcher\EventDispatcherInterface;
 
 class FeedService
 {
-    public function __construct(private EventDispatcherInterface $dispatcher, private UrlGeneratorInterface $urlGenerator)
+    public function __construct(private iterable $feedTypes, private EventDispatcherInterface $dispatcher, private UrlGeneratorInterface $urlGenerator)
     {
     }
 
     public function getFeedTypes(): array
     {
-        $event = new GetFeedTypesEvent();
-        $event = $this->dispatcher->dispatch($event, GetFeedTypesEvent::NAME);
+        $res = [];
 
-        return $event->getFeedTypes();
+        foreach ($this->feedTypes as $feedType) {
+            $res[] = $feedType::class;
+        }
+
+        return $res;
     }
 
     public function getFeedUrl(Feed $feed): string
     {
+        // @TODO: Find solution without depending on @internal RouteNameGenerator.
```

Comment:
https://symfony.com/doc/current/routing.html#generating-urls-in-services

Have you tried switching  `use ApiPlatform\Core\Api\UrlGeneratorInterface;` -> `use Symfony\Component\Routing\Generator\UrlGeneratorInterface;`